### ISSUE TITLE
Basic Pipelined-Rendering App has Excessive CPU Utilization

### ISSUE THREAD

**Issue by Supermath101** [bevyengine/bevy#3210]
## Bevy version

Commit ID: `e8412df0`

## Operating system & version

Manjaro Linux with Nvidia proprietary drivers.

## What you did

I ran a very basic Pipelined rendering app on release mode.

```rs
use bevy::{
    ecs::prelude::*,
    prelude::{App},
    PipelinedDefaultPlugins,
    render2::{
        camera::OrthographicCameraBundle
    }
};

fn main() {
    App::new().add_plugins(PipelinedDefaultPlugins)
    .add_startup_system(setup_system)
    .run();
}

fn setup_system(mut commands: Commands) {
    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
}
```

## What you expected to happen

This example code should in-theory run at near 0% CPU utilization.

## What actually happened

The CPU utilization was hovering around 1/3 usage.

## Additional information

The output of running `cargo-flamegraph`: [flamegraph.zip](https://github.com/bevyengine/bevy/files/7612810/flamegraph.zip)

I'm running an [Intel Core i7-9750H](https://www.intel.com/content/www/us/en/products/sku/191045/intel-core-i79750h-processor-12m-cache-up-to-4-50-ghz/specifications.html) with 16 GB of RAM.

This was the 2nd item that was logged (I thought it might be useful):
```
INFO bevy_render2::renderer: AdapterInfo { name: "NVIDIA GeForce GTX 1660 Ti", vendor: 4318, device: 8593, device_type: DiscreteGpu, backend: Vulkan }
```

**Comment by alice-i-cecile** [comment]
Per Discord, this still happens on the old renderer, and on 0.5, correct?

**Comment by alice-i-cecile** [comment]
For reference: I'm getting about 1-2% CPU usage with a very basic scene on the same commit.

**Comment by Supermath101** [comment]
> Per Discord, this still happens on the old renderer, and on 0.5, correct?

Correct.

**Comment by superdump** [comment]
I’m wondering if somehow vsync is turned off.

**Comment by cart** [comment]
Yup this is almost certainly due to a lack of frame limiting. Right now vsync is our only default form of frame limiting, which isn't great for a variety of reasons:

1. Games without vsync munch on CPUs
2. We can't use the (often better) Mailbox vsync mode where it is available without spiking CPU usage

On top of that, we'll need to sort out a new way to handle "delta times" anyway when we enable "actual parallel pipelining" in the new renderer, which im still working on. I might just add "on by default" frame limiting with those changes, as I think the changes I need to make have lots of overlap with frame limiting.

**Comment by maniwani** [comment]
@TheRawMeatball and I discussed frame-rate limiting before (inspired by the user who hit the change detection warning overnight). Seems like it'd be a small change to the winit loop. Relevant discussion. <https://discord.com/channels/691052431525675048/743663924229963868/906440620640452650>
